packages/cli/src/lib/bundler/config.ts

```diff
@@ -232,12 +233,50 @@ export async function createConfig(
     require.resolve('react-refresh'),
   ];
 
+  const mode = isDev ? 'development' : 'production';
+  const optimization = optimizationConfig(options);
+
+  if (
+    process.env.EXPERIMENTAL_MODULE_FEDERATION === 'true' &&
+    process.env.FORCE_REACT_DEVELOPMENT === 'true'
+  ) {
+    console.log(
+      chalk.yellow(
+        `⚠️  WARNING: Forcing react and react-dom into development mode. This build should not be used in production.`,
+      ),
+    );
+
+    const reactPackageDirs = [
+      `${dirname(require.resolve('react/package.json'))}/`,
+      `${dirname(require.resolve('react-dom/package.json'))}/`,
+    ];
+
+    // Don't define process.env.NODE_ENV with value matching config.mode.
```

Comment:
I think it worth noting here that if we don't set this, the below `runtimeValue` will be ignored

Reply:
:+1: 0983429447db